src/views/Rules/ListRules/index.jsx

```diff
@@ -258,16 +258,18 @@ function ListRules(props) {
   }, []);
 
   useEffect(() => {
-    fetchRoles(username).then(userInfo => {
-      const roleList =
-        (userInfo.data && Object.keys(userInfo.data.data.roles)) || [];
+    if (username !== '') {
```

Comment:
If we ever we decide to change the fallback value
https://github.com/mozilla-frontend-infra/balrog-ui/blob/bb3410c401b3f20b39def7526922ef8b64a9daba/src/views/Rules/ListRules/index.jsx#L74

to something like `null` instead of `''`, then this piece of code may break. We could make this more resilient with `if (username) {`

Let me know if you had different thoughts.

Reply:
Makes sense to me. I started avoiding bare `if`s after learning that they don't work as expected for Objects, but it totally makes sense here.